bee-storage/bee-storage/src/health.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines a type to represent different health states in which the storage backend can be.
+
+use bee_common::packable::{Packable, Read, Write};
+
+/// Errors related to storage health statuses.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// I/O error happened.
+    #[error("I/O error happened: {0:?}")]
+    Io(#[from] std::io::Error),
+    /// Unknown storage health variant.
+    #[error("Unknown storage health variant: {0}")]
+    UnknownHealthVariant(u8),
+}
+
+/// Represents different health states for a `StorageBackend`.
+#[repr(u8)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum StorageHealth {
+    /// Healthy backend state.
+    Healthy = 0,
+    /// Idle backend state.
+    Idle = 1,
+    /// Corrupted backend state.
+    Corrupted = 2,
```

Comment:
The doc comments are a bit useless, because we are basically just repeating the variant's name, although they are not self-explanatory. We should mention what we consider a healthy, idle, and corrupted, and why they are mutually exclusive.